### PR TITLE
[Connector API] Null request body should result in 400 status code when body is required

### PR DESCRIPTION
**Pull request by jedrazb**
### Problem

Subset of Connector APIs handlers assumes that the request body is always provided. The body parameter is marked as `required` in the API spec (used to generate clients), but there is nothing stopping a user from sending a request without body.

If body is not provided for a request where body is expected, we are failing with a 500 error, as the server fails at processing the body that is `null`. This is not a good way to handle such scenario. 

### Changes

Looking at other API endpoint([1](https://github.com/elastic/elasticsearch/blob/4a0b026276613cd48895e3b35cb266a2abdcfa12/modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/rest/RestPutDataStreamLifecycleAction.java#L43), [2](https://github.com/elastic/elasticsearch/blob/4a0b026276613cd48895e3b35cb266a2abdcfa12/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/secrets/action/RestPostConnectorSecretAction.java#L36), [3](https://github.com/elastic/elasticsearch/blob/4a0b026276613cd48895e3b35cb266a2abdcfa12/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/saml/RestSamlCompleteLogoutAction.java#L76)) I took inspiration and followed their pattern to gracefully handle missing body param. If body is not provided for a request that expects a body, we return informative message and 400 status code. See example:

```
{
    "error": {
        "root_cause": [
            {
                "type": "parse_exception",
                "reason": "request body is required"
            }
        ],
        "type": "parse_exception",
        "reason": "request body is required"
    },
    "status": 400
}
```

I'm using `request.contentParser()` - empty body would cause an exception that would be propagated as a 400 status code in the API response (like above).

I removed a bunch of unnecessary `fromXContentBytes` functions, in favour of keeping it simpler.

I adapted all Connector API endpoints that require the body.

### Testing
- I tried adding yaml e2e tests for those scenarios but I'm getting errors from api spec if I intentionally skip body parameter in the request (so can't test this scenario there)
- No effects on existing logic, so no tests changed
- Verified locally that this works as expected

### CODE REVIEW

**Comment by jedrazb** [comment]
@elasticmachine merge upstream

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorNameAction.java [hunk 1]
```diff
@@ -13,36 +13,41 @@
...
-        return channel -> client.execute(
-            UpdateConnectorNameAction.INSTANCE,
-            request,
-            new RestToXContentListener<>(channel, ConnectorUpdateActionResponse::status)
-        );
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+        try (XContentParser parser = restRequest.contentParser()) {
+            UpdateConnectorNameAction.Request request = UpdateConnectorNameAction.Request.fromXContent(
+                parser,
+                restRequest.param("connector_id")
```
Comment (seanstory) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorNameAction.java:44]:
```suggestion
                restRequest.param(CONNECTOR_ID_PARAM)
```

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorNativeAction.java [hunk 2]
```diff
@@ -13,36 +13,41 @@
...
-        return channel -> client.execute(
-            UpdateConnectorNativeAction.INSTANCE,
-            request,
-            new RestToXContentListener<>(channel, ConnectorUpdateActionResponse::status)
-        );
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+        try (XContentParser parser = restRequest.contentParser()) {
+            UpdateConnectorNativeAction.Request request = UpdateConnectorNativeAction.Request.fromXContent(
+                parser,
+                restRequest.param("connector_id")
```
Comment (seanstory) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorNativeAction.java:44]:
```suggestion
                restRequest.param(CONNECTOR_ID_PARAM)
```

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorPipelineAction.java [hunk 3]
```diff
@@ -13,36 +13,41 @@
...
-        return channel -> client.execute(
-            UpdateConnectorPipelineAction.INSTANCE,
-            request,
-            new RestToXContentListener<>(channel, ConnectorUpdateActionResponse::status)
-        );
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+        try (XContentParser parser = restRequest.contentParser()) {
+            UpdateConnectorPipelineAction.Request request = UpdateConnectorPipelineAction.Request.fromXContent(
+                parser,
+                restRequest.param("connector_id")
```
Comment (seanstory) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorPipelineAction.java:44]:
```suggestion
                restRequest.param(CONNECTOR_ID_PARAM)
```

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorSchedulingAction.java [hunk 4]
```diff
@@ -13,36 +13,41 @@
...
-        return channel -> client.execute(
-            UpdateConnectorSchedulingAction.INSTANCE,
-            request,
-            new RestToXContentListener<>(channel, ConnectorUpdateActionResponse::status)
-        );
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+        try (XContentParser parser = restRequest.contentParser()) {
+            UpdateConnectorSchedulingAction.Request request = UpdateConnectorSchedulingAction.Request.fromXContent(
+                parser,
+                restRequest.param("connector_id")
```
Comment (seanstory) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorSchedulingAction.java:44]:
```suggestion
                restRequest.param(CONNECTOR_ID_PARAM)
```

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorServiceTypeAction.java [hunk 5]
```diff
@@ -13,36 +13,41 @@
...
-        return channel -> client.execute(
-            UpdateConnectorServiceTypeAction.INSTANCE,
-            request,
-            new RestToXContentListener<>(channel, ConnectorUpdateActionResponse::status)
-        );
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+        try (XContentParser parser = restRequest.contentParser()) {
+            UpdateConnectorServiceTypeAction.Request request = UpdateConnectorServiceTypeAction.Request.fromXContent(
+                parser,
+                restRequest.param("connector_id")
```
Comment (seanstory) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorServiceTypeAction.java:44]:
```suggestion
                restRequest.param(CONNECTOR_ID_PARAM)
```

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorStatusAction.java [hunk 6]
```diff
@@ -13,36 +13,41 @@
...
-        return channel -> client.execute(
-            UpdateConnectorStatusAction.INSTANCE,
-            request,
-            new RestToXContentListener<>(channel, ConnectorUpdateActionResponse::status)
-        );
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+        try (XContentParser parser = restRequest.contentParser()) {
+            UpdateConnectorStatusAction.Request request = UpdateConnectorStatusAction.Request.fromXContent(
+                parser,
+                restRequest.param("connector_id")
```
Comment (seanstory) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorStatusAction.java:44]:
```suggestion
                restRequest.param(CONNECTOR_ID_PARAM)
```

**Review by seanstory — COMMENTED** [review]
> I tried adding yaml e2e tests for those scenarios but I'm getting errors from api spec if I intentionally skip body parameter in the request (so can't test this scenario there)

Would unit tests work, instead of yaml tests?

It's not clear to me (a novice) why these changes fix the issue you describe, so without test changes, I have to just take your word for it that this makes a difference. And i do trust you, but I don't trust that we won't see regressions if we don't have a way to test this.

**Comment by jedrazb** [comment]
> Would unit tests work

Let me look into this!

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorErrorAction.java [hunk 7]
```diff
@@ -13,36 +13,41 @@
...
-        return channel -> client.execute(
-            UpdateConnectorErrorAction.INSTANCE,
-            request,
-            new RestToXContentListener<>(channel, ConnectorUpdateActionResponse::status)
-        );
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+        try (XContentParser parser = restRequest.contentParser()) {
+            UpdateConnectorErrorAction.Request request = UpdateConnectorErrorAction.Request.fromXContent(
+                parser,
+                restRequest.param("connector_id")
```
[the remark this answers is not in the document]
Comment (jedrazb) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorErrorAction.java:44]: Hmm, weird. It seems my intellij local index got into weird state, I was sure I replaced all occurrences of it (searching within `action` module for connector_id returns no results) ... Anyway, thank you for flagging!

**Comment by jedrazb** [comment]
> It's not clear to me (a novice) why these changes fix the issue you describe, so without test changes, I have to just take your word for it that this makes a difference. 

@seanstory You are totally right saying that, I included unit tests for all endpoints that I modified. That required adding some new files (resulting in a lot of additions), but the changes are pretty straightforward

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorErrorAction.java [hunk 8]
```diff
@@ -13,36 +13,41 @@
...
 
+import java.io.IOException;
 import java.util.List;
 
 import static org.elasticsearch.rest.RestRequest.Method.PUT;
 
 @ServerlessScope(Scope.PUBLIC)
 public class RestUpdateConnectorErrorAction extends BaseRestHandler {
 
+    private static final String CONNECTOR_ID_PARAM = "connector_id";
```
[the remark this answers is not in the document]
Comment (jedrazb) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorErrorAction.java:27]: Sounds like a refactoring, I filed the issue to look into this https://github.com/elastic/search-team/issues/7790 this PR got quite big already

**Comment by jedrazb** [comment]
@elasticmachine merge upstream

**Review by seanstory — APPROVED** [review]
nice!

**Comment by jedrazb** [comment]
@elasticmachine merge upstream

**Pull request merged**